key.go

```diff
@@ -0,0 +1,69 @@
+package statter
+
+import (
+	"sync"
+	"unsafe"
+)
+
+var keyPool = sync.Pool{
+	New: func() interface{} {
+		return &key{b: make([]byte, 0, 256)}
+	},
+}
+
+type key struct {
+	b []byte
+}
+
+func newKey(name string, tags []Tag) *key {
+	k := keyPool.Get().(*key)
+	k.b = k.b[:0]
+
+	k.b = append(k.b, name...)
+
+	// Short path for no tags.
+	if len(tags) == 0 {
+		return k
+	}
+
+	// The tags must be sorted to create a consistent key.
+	sortTags(tags)
+
+	for _, tag := range tags {
+		k.b = append(k.b, ':')
+		k.b = append(k.b, tag[0]...)
+		k.b = append(k.b, '=')
+		k.b = append(k.b, tag[1]...)
```

Comment:
It would probably make sense to have the `:` and `=` characters in consts that make their role explicit, WDYT?